File: site_frontend/lib/data/event.dart
class Event {
  String id;
  String name;
  String description;
  String category;
  String city;
  double price;
  DateTime date;
  String imageUrl;
  String location;
  int capacity;

  Event({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.city,
    required this.price,
    required this.date,
    required this.imageUrl,
    required this.location,
    required this.capacity,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'].toString(),
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? '',
      city: json['city'] ?? '',
      price: (json['price'] as num).toDouble(),
      date: DateTime.parse(json['date']),
      imageUrl: json['imageUrl'] ?? '',
      location: json['location'] ?? '',
      capacity: json['capacity'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'category': category,
      'city': city,
      'price': price,
      'date': date.toIso8601String(),
      'imageUrl': imageUrl,
      'location': location,
      'capacity': capacity,
    };
  }
}

File: site_backend/Models/Event.cs
using System.ComponentModel.DataAnnotations;

namespace site_backend.Models
{
    public class Event
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public string Category { get; set; } = null!;
    public string City { get; set; } = null!;
    public double Price { get; set; }
    public DateTime Date { get; set; }
    public string ImageUrl { get; set; } = null!;
    public string Location { get; set; } = null!;
    public int Capacity { get; set; }

    public int UserId { get; set; }

    // Navigation property optional yap
    public User? User { get; set; }
}

}
